ltable.h:12:15: error: static_assert expression is not an integral constant expression

It looks like my compiler is having a problem with this code:

```cpp
static_assert(offsetof(LuaNode, val) == 0, "Unexpected Node memory layout, pointer cast below is incorrect");
```

It gives these errors:
```
  ltable.h:12:15: error: static_assert expression is not an integral constant expression
  ltable.h:12:15: note: cast that performs the conversions of a reinterpret_cast is not allowed in a constant expression
  ltable.h:12:15: error: static_assert expression is not an integral constant expression
```

Due to the build process being magical, I'm not sure exactly which compiler is being used to compile this. Probably MSVC, but I don't know which one (because Windows is Windows).

--- a/VM/src/ltable.cpp
+++ b/VM/src/ltable.cpp
@@ -30,6 +30,7 @@ LUAU_FASTFLAGVARIABLE(LuauArrayBoundary, false)
 #define MAXBITS 26
 #define MAXSIZE (1 << MAXBITS)
 
+static_assert(offsetof(LuaNode, val) == 0, "Unexpected Node memory layout, pointer cast in gval2slot is incorrect");
 // TKey is bitpacked for memory efficiency so we need to validate bit counts for worst case
 static_assert(TKey{{NULL}, 0, LUA_TDEADKEY, 0}.tt == LUA_TDEADKEY, "not enough bits for tt");
 static_assert(TKey{{NULL}, 0, LUA_TNIL, MAXSIZE - 1}.next == MAXSIZE - 1, "not enough bits for next");

--- a/VM/src/ltable.h
+++ b/VM/src/ltable.h
@@ -9,7 +9,6 @@
 #define gval(n) (&(n)->val)
 #define gnext(n) ((n)->key.next)
 
-static_assert(offsetof(LuaNode, val) == 0, "Unexpected Node memory layout, pointer cast below is incorrect");
 #define gval2slot(t, v) int(cast_to(LuaNode*, static_cast<const TValue*>(v)) - t->node)
 
 LUAI_FUNC const TValue* luaH_getnum(Table* t, int key);

